services/viva/api/cases/test/lambdas/addCasePerson.test.ts

```diff
@@ -0,0 +1,165 @@
+import type { Token } from '../../src/libs/token';
+import { addCasePerson } from '../../src/lambdas/addCasePerson';
+import { CasePerson, CasePersonRole, EncryptionType } from '../../src/types/caseItem';
+import type { CaseItem, CaseForm } from '../../src/types/caseItem';
+import type { LambdaRequest, Dependencies } from '../../src/lambdas/addCasePerson';
+
+const personalNumber = '199801011212';
+const SK = 'CASE#123';
+const PK = `USER#${personalNumber}`;
+const caseKeys = { PK, SK };
+
+const coApplicant: CasePerson = {
+  personalNumber: '199701031212',
+  firstName: 'Svenne',
+  lastName: 'Banan',
+  hasSigned: false,
+  role: CasePersonRole.CoApplicant,
+};
+
+const form: CaseForm = {
+  answers: [
+    {
+      field: {
+        id: 'some id 123',
+        tags: [],
+      },
+      value: 'Some value abc',
+    },
+  ],
+  currentPosition: {
+    currentMainStep: 0,
+    currentMainStepIndex: 0,
+    index: 0,
+    level: 0,
+  },
+  encryption: {
+    type: EncryptionType.Decrypted,
+  },
+};
+
+const caseItem: CaseItem = {
+  PK,
+  SK,
+  id: '123',
+  createdAt: 0,
+  currentFormId: '123abc',
+  details: {
+    period: {
+      startDate: 0,
+      endDate: 0,
+    },
+    workflowId: '123',
+  },
+  state: 'SOME STATE',
+  expirationTime: 0,
+  updatedAt: 0,
+  status: {
+    type: '',
+    name: '',
+    description: '',
+  },
+  forms: {
+    '123abc': form,
+  },
+  provider: 'VIVA',
+  persons: [coApplicant],
+};
+
+function createInput(partialInput: Partial<LambdaRequest> = {}): LambdaRequest {
+  return {
+    body: JSON.stringify({
+      personalNumber: '199701031212',
+      firstName: 'Svenne',
+      lastName: 'Banan',
+    }),
+    pathParameters: {
+      caseId: '123',
+    },
+    headers: {
+      Authorization: `Bearer ${personalNumber}`,
+    },
+    ...partialInput,
+  };
+}
+
+function createDependencies(
+  tokenToUse: Token,
+  caseItemToUse: CaseItem,
+  partialDependencies: Partial<Dependencies> = {}
+): Dependencies {
+  return {
+    decodeToken: () => tokenToUse,
+    updateCase: () => Promise.resolve({ Attributes: caseItemToUse }),
+    getFormTemplates: () => Promise.resolve(),
+    coApplicantStatus: () => Promise.resolve(),
+    validateCoApplicantStatus: () => true,
+    getCase: () => Promise.resolve(caseItemToUse),
+    populateForm: () => ({}),
+    ...partialDependencies,
+  };
+}
+
+it('successfully add person to case', async () => {
+  const updateCaseMock = jest.fn().mockResolvedValueOnce({ Attributes: { ...caseItem } });
+  const getFormTemplatesMock = jest.fn().mockResolvedValueOnce({});
+
+  const input = createInput();
+  const dependencies = createDependencies(
+    { personalNumber },
+    { ...caseItem },
+    {
+      updateCase: updateCaseMock,
+      getFormTemplates: getFormTemplatesMock,
+    }
+  );
```

Comment:
Do we really need to send in the caseItem and token as parameters to `createDependencies` function? I mean, if we want another caseItem or token, then we just override that particular function.

like:

```js
const mockToken = {
  personalNumber: '1234',
};

const caseItem = { ... };


function createDependencies(
  partialDependencies: Partial<Dependencies> = {}
): Dependencies {
  return {
    decodeToken: () => mockToken,
    updateCase: () => Promise.resolve({ Attributes: caseItem }),
    getFormTemplates: () => Promise.resolve(),
    coApplicantStatus: () => Promise.resolve(),
    validateCoApplicantStatus: () => true,
    getCase: () => Promise.resolve(caseItem),
    populateForm: () => ({}),
    ...partialDependencies,
  };
}

it('myTest', () => {
  const dependencies = createDependencies(); // Gives default dependencies
}

```

And then if you want to change the caseItem/or token to be returned:

```js
it('mySecondTest', () => {
const myOtherTestToken = {
  personalNumber: '4567',
};
  const dependencies = createDependencies({
  decodeToken: () => myOtherTestToken,
});
});
```